lib/commentOnOpenedPullRequest.js

```diff
@@ -159,24 +163,58 @@ commentOnOpenedPullRequest._askAboutTests = function (files, unitTestPath) {
     return true;
 };
 
-commentOnOpenedPullRequest._askForCla = function (username, repositorySettings) {
-    if (!defined(repositorySettings.claUrl)) {
+commentOnOpenedPullRequest._askForCla = function (userName) {
+    if (!defined(Settings.googleSheetsApi)) {
         return Promise.resolve(false);
     }
 
-    return requestPromise.get({
-        url: repositorySettings.claUrl,
-        headers: repositorySettings.headers,
-        json: true
-    })
-        .then(function (response) {
-            var cla = JSON.parse(Buffer.from(response.content, 'base64').toString());
-            for (var i = 0; i < cla.length; i++) {
-                var claUsername = cla[i].gitHub;
-                if (defined(claUsername) && claUsername === username) {
-                    return false;
+    var deferred = deferredPromise();
+
+    // Check individual CLA sheet.
+    Settings.googleSheetsApi.spreadsheets.values.get({
+        spreadsheetId: Settings.individualClaSheetID,
+        range: 'D2:D'
+    }, function(error, response) {
+        if (error) {
+            deferred.reject('Google Sheets API error: ' + error);
+        }
```

Comment:
There's no return here so the below code runs anyway and probably crashes the server.